content/en/docs/concepts/storage/volumes.md

```diff
@@ -574,10 +574,17 @@ backoff and will be reported on the pod reason and message.
 
 The types of objects that may be mounted by this volume are defined by the
 container runtime implementation on a host machine and at minimum must include
-all valid types supported by the container image field. The OCI object gets
-mounted in a single directory (`spec.containers[*].volumeMounts.mountPath`) by
-will be mounted read-only. On Linux, the container runtime typically also mounts the 
-volume with file execution blocked (`noexec`).
+all valid types supported by the container image field. 
+
+{{< note >}}
+When using the container runtime
+[CRI-O](/docs/setup/production-environment/container-runtimes/#cri-o), the OCI
+object gets mounted read-only into a single directory
+(`spec.containers[*].volumeMounts.mountPath`). On Linux, the container runtime
+typically also mounts the volume with file execution blocked (`noexec`). When
+using SELinux, image volumes cannot be shared across pods applying different
+labels on a single node.
+{{< /note >}}
```

Comment:
Does all the cri implementations have the same limitation, or does it depend on the implementation?

Reply:
If this is a bug in CRI-O, we should frame this slightly differently.

If this is a bug in Kubernetes that only shows up with CRI-O, let's document it as a bug.

Reply:
/hold
for https://github.com/kubernetes/kubernetes/pull/126806#discussion_r1723119109

Reply:
@saschagrunert can you leave a comment about @sftim's question? 

And feel free to cancel the hold if it is ready to be merged.

Reply:
It's not really a bug, more a known limitation.

@haircommander PTAL, should we frame this differently?

Reply:
this is an intentional decision we made as it matches current volume behavior (it was discussed offline in a convo with @gnufied so I can't link unfortunately). I agree with known limitation, or maybe even design decision that some users need to react to. it's not unique to cri-o, it's just that cri-o often runs on nodes that run SELinux

Reply:
OK, if it's generic to all? most? container runtimes on Linux then we should document it as that, and not mention CRI-O.

Reply:
/hold

@haircommander @saschagrunert  sig-storage meeting: https://docs.google.com/document/d/1-8KEG8AjAgKznS9NFm3qWqkGyCHmvU6HVl0sk5hwoAE/edit#heading=h.9rsna5zrjwc

@gnufied will also comment on this issue.

Reply:
Although for most volume plugins it is true that, they can't be mounted with 2 different selinux context on same node, we want to understand why is this not possible for `Imagevolume` source. For a traditional storage persistent storage plugins - this limitation is underlying limitation of storage backend/Linux kernel etc. 

For `ImageVolume` source, it seems like a simple solution which unpacks image to two different hostpath would work and not run into this limitation. @haircommander can you specify, why this limitation exists for `ImageVolume` plugins? 

Having said that - website update is probably not the right place to discuss this and we should have discussed this in enhancement.

Reply:
we choose to keep it simple and follow the lead of other volume types. supporting multiple selinux types would mean making more complicated the way we mount the image into the container AFAIR. But agreed we can chat about this elsewhere, in the meantime for the purposes of this PR: this is consistent with behavior of other volume plugins and is expected behavior

Reply:
> But agreed we can chat about this elsewhere,

created an issue  https://github.com/kubernetes/enhancements/issues/4804 @haircommander @gnufied